apis/v1/shared_types.go

```diff
@@ -535,6 +535,19 @@ type Hostname string
 // +kubebuilder:validation:Pattern=`^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$`
 type PreciseHostname string
 
+// AbsoluteURI represents a Uniform Resource Identifier (URI) as defined by RFC3986.
+
+// The AbsoluteURI MUST NOT be a relative URI, and it MUST follow the URI syntax and
+// encoding rules specified in RFC3986.  The AbsoluteURI MUST include both a
+// scheme (e.g., "http" or "spiffe") and a scheme-specific-part.  URIs that
+// include an authority MUST include a fully qualified domain name or
+// IP address as the host.
```

Comment:
Not sure how we should approach for case-sensitivity, scheme and host aren't but other parts can be. Let me know if you want to add any language to indicate what we expect

Reply:
Maybe "implementations SHOULD treat this string case insensitively" or something? Feels like it's better to have _some_ stance on case sensitivity rather than none.

Reply:
I think I'd leave case sensitivity out of this as it's potentially different depending on the scheme. Per RFC-3986:

>When a URI uses components of the generic syntax, the component syntax equivalence rules always apply; namely, that the scheme and host are case-insensitive and therefore should be normalized to lowercase.  For example, the URI <HTTP://www.EXAMPLE.com/> is equivalent to <http://www.example.com/>.  The other generic syntax components are assumed to be case-sensitive unless specifically defined otherwise by the scheme

https://datatracker.ietf.org/doc/html/rfc3986#section-6.2.2.1